py4ai/core/data/layer/v1/mongo/serializer.py

```diff
@@ -0,0 +1,85 @@
+from hashlib import md5
+
+# from typing import Any
+
+from bson import ObjectId
+from pydantic.main import ModelMetaclass
+
+# from py4ai.core.data.layer.v1.common.serialiazer import DataSerializer
+
+
+def create_mongo_id(key: str) -> ObjectId:
+    return ObjectId(md5(str(key).encode("utf-8")).hexdigest()[:24])
+
+
+class MongoModel(ModelMetaclass):
+    def __init__(cls, name, bases, dct):  # type: ignore
```

Comment:
Why this `type: ignore`?

Reply:
Didn't manage to get the type inference working in Metaclasses